### DISCUSSION TITLE
Supported NodeJS versions

### DISCUSSION THREAD

**Discussion by dantefromhell** [TiddlyWiki/TiddlyWiki5#7012] [General]
Looking at the documentation on how to run tw5 within nodejs I wonder if there are limitations/ supported/ know nodejs:tw5 issues?

**Comment by pmario** [comment]
I do use the latest LTS version 18.x. But node v 16.xx should be more than capable to work with TW

**Reply by dubiouscript** [reply]
i was reminded of this question
>wonder if there are limitations/ supported/ know nodejs:tw5 issue

reading through https://github.com/Jermolene/TiddlyWiki5/pull/5853#issuecomment-877398715
>they changed this timeout to 0 in node 13. ... So if you use a later node version you don't see the problem.

as always the dev is in the detail :bug:

**Comment by Jermolene** [comment]
Hi @dantefromhell – I'd like to keep TW5 compatible with the broadest possible range of Node.js versions, and so I try to avoid dependencies on recent versions. But there is no published policy, nor any explicit testing.

It would be helpful to develop compatibility policies both for Node.js and the browser. In the case of browsers, I'd be thinking in terms of supporting the last five years of browser releases (which now includes async/await, handily). I'm not sure if that would be an appropriate period for Node.js.
